src/test/ui/async-await/generator-not-future.rs

```diff
@@ -0,0 +1,31 @@
+// edition:2018
+#![feature(generators, generator_trait)]
+
+use std::future::Future;
+use std::ops::Generator;
+
+async fn func() {}
+
+fn takes_future(_f: impl Future<Output = ()>) {}
+fn takes_generator<ResumeTy>(_g: impl Generator<ResumeTy, Yield = (), Return = ()>) {}
+
+fn main() {
+    // okay:
+    takes_future(func());
+    takes_future(async {});
+    takes_generator(|| {
```

Comment:
To make this more parallel, can you also do a `takes_generator(func2())` or something?

Reply:
I guess it doesn't really matter b/c we can't name a generator type in the same way as an `async fn`.